packages/ui/src/Common.ts

```diff
@@ -1449,29 +1494,56 @@ export interface InfoTooltipButtonProps {
 }
 
 export interface ModalProps {
-  onDismiss: () => void;
-  visible: boolean;
-  // Alignment of the header. Default is "center".
-  align?: "center" | "start";
-  // Element to render in the middle part of the modal.
+  /**
+   * The content of the modal.
+   */
   children?: React.ReactElement;
-  // Element to render in the bottom of the modal. This takes precedence over primaryButton and
-  // secondaryButton.
-  footer?: React.ReactElement;
-  heading?: string;
-  size?: "sm" | "md" | "lg";
-  subHeading?: string;
-  // Renders a primary colored button all the way to the right in the footer, if no footer prop is
-  // provided.
-  primaryButtonText?: string;
-  primaryButtonOnClick?: (value?: any) => void;
+  /**
+   * If true, the primary button will be disabled.
+   * @default false
+   */
   primaryButtonDisabled?: boolean;
-  // Renders a gray button to the left of the primary button in the footer, if no footer prop is
-  // provided. Requires primaryButtonText to be defined, but is not required itself.
+  /**
+   * The text content of the primary button.
+   */
+  primaryButtonText?: string;
+  /**
+   * The text content of the secondary button.
+   */
   secondaryButtonText?: string;
+  /**
+   * The size of the modal.
+   * @default "sm"
+   */
+  size?: "sm" | "md" | "lg";
+  /**
+   * The subtitle of the modal.
+   */
+  subTitle?: string;
+  /**
+   * The text content of the modal.
+   */
+  text?: string;
+  /**
+   * The title of the modal.
+   */
+  title?: string;
+  /**
+   * If true, the modal will be visible.
+   */
+  visible: boolean;
+  /**
+   * The function to call when the modal is dismissed.
+   */
+  onDismiss: () => void;
+  /**
+   * The function to call when the primary button is clicked.
+   */
+  primaryButtonOnClick?: (value?: any) => void;
```

Comment:
Same here, probably should be void | Promise<void> and wait to dismiss if there's an async action.